src/main/java/com/minecolonies/coremod/colony/CitizenData.java

```diff
@@ -453,12 +453,19 @@ public void initEntityValues()
         // Safety check that ensure the citizen its job matches the work building job
         if (getJob() != null && workBuilding != null)
         {
-            try
+            boolean hasCorrectJob = false;
+            for (WorkerBuildingModule module : workBuilding.getModules(WorkerBuildingModule.class))
             {
-                workBuilding.getModuleMatching(WorkerBuildingModule.class, m -> m.getJobEntry().equals(getJob().getJobRegistryEntry()));
+                if (module.getJobEntry().equals(getJob().getJobRegistryEntry()))
+                {
+                    hasCorrectJob = true;
+                    break;
+                }
             }
-            catch (Exception ex)
+
+            if (!hasCorrectJob)
             {
+                workBuilding.getModuleMatching(WorkerBuildingModule.class, m -> m.getJobEntry().equals(getJob().getJobRegistryEntry()));
```

Comment:
Wouldn't this line throw an exception when `hasCorrectJob == false`?

Reply:
Good catch, that's not supposed to be there anymore